File: python/structs/ArrayList.py
class ArrayList:
  def __init__(self):
    self.list = []
    self.size = 0
  def add(self, e):
    self.list[self.size] = e
    self.size += 1
  def contains(self, e):
    for el in self.list:
      if(el == e):
        return True
    return False
  def clear(self):
    self.list = []
    self.size = 0
  def remove(self, index):
    if(index < 0 or self.size == 0 or index > self.size):
      raise IndexError
    temp = self.list[index]
    for i in range(index, len(list) - 1):
      self.list[i] = self.list[i + 1]
    self.size -= 1
    return temp
  def get(self, index):
    if(index < 0 or size == 0 or index > self.size):
      raise IndexError
    return self.list[index]
  

File: php/structs/ArrayList.php
<?php

class ArrayList
{
    private $data;

    public function __construct()
    {
        $this -> data = [];
    }

    public function add(string $item)
    {
        $this -> data[] = $item;
    }

    public function contains(string $item)
    {
        for ($i = 0; $i < sizeof($this -> data); $i++) {
            if ($this -> data[$i] == $item) {
                return true;
            }
        }
        return false;
    }

    public function remove(int $index){
      if($index < 0 || $index > sizeof($this -> data) || sizeof($this -> data) ==0){
        return -1;
      }
      $value = $this -> data[$index];
      unset($this -> data[$index]);
      $this -> data = array_values($this -> data);
      return $value;
    }

    public function get(int $index){
      if($index < 0 || $index > sizeof($this -> data) || sizeof($this -> data) == 0) {
        return -1;
      }
      return $this -> data[$index];
    }
}
